### ISSUE TITLE
`Popup`: issues with children controls on custom widget

### ISSUE THREAD

**Issue by rblopes** [godotengine/godot#72850]
### Godot version

v4.0.beta16.official [518b9e580]

### System information

KDE neon 22.04 (Ubuntu-derived)

### Issue description

I'm migrating one of my projects to Godot 4.0. There, I made a filtered list popup widget, using `Popup` as its root node.

Right now, I'm having the following issues:
- Calling `grab_focus` on the `LineEdit` child node hides the popup immediately, it needs to be shown a second time to become usable;
- Children controls cannot be interacted with the mouse, the popup is hidden whenever I click or use the scroll wheel.

These issues only occur when "Embed Subwindows" (`display/window/subwindows/embed_subwindows`) are enabled on the project settings, regardless of the renderer chosen.

I'm not sure if something in that widget's node setup is causing the issue, however that same widget works with no issues on the former Godot 3.5 project.

Also tested with beta.15 and beta.17 with same results.

### Steps to reproduce

Please refer to the MRP for more details.

### Minimal reproduction project
[popup-issues-mrp.zip](https://github.com/godotengine/godot/files/10679820/popup-issues-mrp.zip)

_(Edit: updated MRP to not override `grab_focus`.)_

**Comment by Sauermann** [comment]
When I try to run the mrp in self compiled v4.0.beta.custom_build [bdad9770d], I get the following error:
```
Parser Error: The method "grab_focus" overrides a method from native class "Control". This won't be called by the engine and may not work as expected. (Warning treated as error.)
```
This message doesn't appear when I run it in Godot 4.0 Beta 17.

**Comment by YuriSizov** [comment]
@Sauermann This is correct. Shadowing of native methods is not going to work, so we now have a warning/error about it.

**Comment by rblopes** [comment]
Ok, I've updated the MRP to remove that `grab_focus` override. Not sure if using `visibility_changed` is the way to go, but should work for now.

**Comment by Sauermann** [comment]
Can confirm the bug in Beta 17.
v4.0.beta.custom_build [bdad9770d] no longer shows this problem.
My guess would be that this got fixed by #72785.
